### DISCUSSION TITLE
How do I pin a specific version of Node when not using bzlmod?

### DISCUSSION THREAD

**Discussion by njlr** [aspect-build/rules_js#1112] [Q&A] [answered]
With `rules_nodejs` I would do this:

```starlark
load("@build_bazel_rules_nodejs//:index.bzl", "node_repositories")

node_repositories(
  node_version = "16.16.0",
  yarn_version = "1.22.19",
)

load("@rules_nodejs//nodejs:repositories.bzl", "nodejs_register_toolchains")

nodejs_register_toolchains(
  name = "nodejs",
  node_version = "16.16.0",
)
```

What is this the right approach for `rules_js`?

**Comment by jbedard** [comment, marked as answer]
The node toolchains still come from rules_nodejs, see the WORKSPACE snippets in the release notes: https://github.com/aspect-build/rules_js/releases

**Comment by paladox** [comment]
Hi, what's the replacement for yarn_install (or rather yarn_version)? I'm not seeing any in rule_js.

**Reply by jbedard** [reply]
pnpm with [npm_translate_lock](https://github.com/aspect-build/rules_js/blob/main/docs/npm_translate_lock.md), optionally by importing a yarn.lock file to convert to pnpm

**Reply by paladox** [reply]
Oh so you can’t use yarn? Does it use a local install of pnpm or does it install it for you like yarn was?

**Reply by jbedard** [reply]
https://github.com/aspect-build/rules_js/blob/main/docs/pnpm.md
